test/common/runtime/runtime_impl_test.cc

```diff
@@ -192,6 +196,14 @@ TEST_F(DiskLoaderImplTest, All) {
   // test_feature_false is not in runtime_features.cc and so is false by default.
   EXPECT_EQ(false, snapshot->runtimeFeatureEnabled("envoy.reloadable_features.test_feature_false"));
 
+#ifdef ENVOY_ENABLE_QUIC
+  EXPECT_EQ(
+      true,
+      snapshot->runtimeFeatureEnabled(
+          "envoy.reloadable_features.FLAGS_quic_reloadable_flag_quic_testonly_default_false"));
+  EXPECT_EQ(true, GetQuicReloadableFlag(quic_testonly_default_false));
```

Comment:
Please add a test that flips the runtime feature to true followed by flipping the runtime feature to false.

What is the expected behavior for QUIC flags in this case?

Also consider the case where you are transitioning from a config that sets a runtime feature that is default false to true via config, and in a future config update the flag is removed from the config.  Does that result in the feature flag default being used after the removal?

Reply:
Great comments!
I digged into the current GFE behaviors, and to answer your questions:
If the flag is flipped back in a later pushed config, the new value should be adopted.
If the flag override is absent in a later pushed config, the flag default value should be restored.

I added new tests in runtime_impl_test for these behaviors.